docker-compose.yml

```diff
@@ -1,109 +1,74 @@
-version: "3.8"
+version: "3.9"
+
+x-var:
+  - &POSTGRES_USER
+    postgres
+  - &POSTGRES_PASSWORD
+    default
+  - &POSTGRES_DATABASE
+    postgres
+  - &POSTGRES_HOST
+    database
 
 services:
   database:
-    container_name: postgres
+    container_name: *POSTGRES_HOST
     environment:
-      POSTGRES_USER: default
-      POSTGRES_PASSWORD: default
-      POSTGRES_DB: default
-    hostname: database
+      POSTGRES_USER: *POSTGRES_USER
+      POSTGRES_PASSWORD: *POSTGRES_PASSWORD
+      POSTGRES_DB: *POSTGRES_DATABASE
     image: postgres:15
-    ports:
-      - "5432:5432"
-    restart: always    
-    volumes:
-      - ./database/init-db.sh:/docker-entrypoint-initdb.d/init-db.sh
-      - /pgdata
+    ports: [5432:5432]
+    volumes: [/pgdata]
     healthcheck:
-      test: [ "CMD-SHELL", "pg_isready", "-d", "db_prod" ]
-      interval: 5s
-      timeout: 10s
-      retries: 10
-    logging:
-      driver: "json-file"
-      options:
-        max-size: 10m
-        max-file: "3"
+      test: ["CMD", "pg_isready", "-U", *POSTGRES_USER]
 
   backend:
     container_name: backend
+    image: maven:3.9.3-eclipse-temurin-17
+    entrypoint: ./mvnw -B verify
+    working_dir: /app
     environment:
-      POSTGRESQL_HOST: database
-      POSTGRESQL_USER: default
-      POSTGRESQL_PASSWORD: default
-      POSTGRESQL_DATABASE: default
+      POSTGRESQL_HOST: *POSTGRES_HOST
+      POSTGRESQL_USER: *POSTGRES_USER
+      POSTGRESQL_PASSWORD: *POSTGRES_PASSWORD
+      POSTGRESQL_DATABASE: *POSTGRES_DATABASE
       CHES_CLIENT_ID: id    
       CHES_CLIENT_SECRET: secret
       CHES_TOKEN_URL: https://dev.loginproxy.gov.bc.ca/auth/realms/comsvcauth/protocol/openid-connect/token
       CHES_API_URL: https://ches-dev.api.gov.bc.ca/api/v1
-    ports:
-      - 3000:3000
+      ENV LANG: en_CA.UTF-8
+      LANGUAGE: en_CA.UTF-8
+      LC_ALL: en_CA.UTF-8
+    ports: [3000:3000]
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:3000/health"]
-      interval: 1m30s
-      timeout: 10s
-      retries: 3
-      start_period: 10s
-    hostname: backend
-    image: ghcr.io/bcgov/nr-forest-client/backend:517
-    depends_on:
-      - database
-    links:
-      - database    
-    logging:
-      driver: "json-file"
-      options:
-        max-size: 10m
-        max-file: "3"
+    depends_on: [*POSTGRES_HOST]
+    volumes: ["./backend:/app"]
 
   frontend:
     container_name: frontend
+    image: node:18-bullseye
+    entrypoint: sh -c "npm ci && npm run start"
```

Comment:
in this case, I think npm install is the best approach for dev environment, and also, I think it should be just `npm start`